python/mxnet/gluon/block.py

```diff
@@ -172,9 +172,23 @@ def __repr__(self):
 
     def __setattr__(self, name, value):
         """Registers parameters."""
+
+        if hasattr(self, name):
+            existing = getattr(self, name)
+            if not name.startswith('_') and not isinstance(value, type(existing)):
+                raise TypeError('Changing attribute type for {name} from {type1} to {type2}' \
```

Comment:
Let's restrict the check to Parameter and Block.

Reply:
Existing attribute value can be NoneType and later changes to Block or Parameter. Should this be allowed?